src/cpuid/src/brand_string.rs

```diff
@@ -104,6 +104,7 @@ impl BrandString {
     /// of the host CPU.
     pub fn from_host_cpuid() -> Result<Self, Error> {
         let mut this = Self::new();
+        // SAFETY: This is safe since the parameters are valid.
```

Comment:
Something along the lines of "0x8000_... is defined as such and such", quoting the CPUID docs?

Reply:
Yes

Reply:
Updated the comment, but in the meantime I realized that there's no clear description of what the unsafety invariants of __cpuid are.
I'll ask in the the Rust zulip.

Reply:
So whether this is safe relates to the specific Intel and AMD CPUID specifications, I expect this is quite a significant amount of work.

Since this is a new comment you are adding if you don't want to dive deep into this specific area, it would be okay to allow the warning here.

Reply:
Looking over the AMD docs references here:
https://doc.rust-lang.org/beta/core/arch/x86_64/fn.__cpuid_count.html

As far as I can tell, 0x8000_0000 is always available; and will return 0x8000_0000 if no other extended functions are supported.

 > The extended functions have numbers in the 8000_xxxxh series (for example, extended
function 8000_0001h). To determine the largest extended function number that a processor supports,
execute CPUID extended function 8000_0000h. If the value returned in EAX is greater than
8000_0000h, the processor supports extended functions.

Hard for me to tell if that's enough of an assurance for the safety of this operation.

Anyway, by "allow" did you mean to modify the comment to remove "SAFETY" and add `#[allow(` above it?

Reply:
> Hard for me to tell if that's enough of an assurance for the safety of this operation.

If the AMD specification and Intel specification on CPUID agree it is always available this would be enough assurance and the comment should contain a reference to where these specification note this.

> Anyway, by "allow" did you mean to modify the comment to remove "SAFETY" and add #[allow( above it?

Yes, if we are not confident in the comment it is better to use `#[allow(clippy::undocumented_unsafe_blocks)]` than to write an inaccurate comment.